services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -1520,6 +1522,60 @@ public void migrateStore(String srcClusterName, String destClusterName, String s
     versionMigrationConsumer.accept(storeName);
   }
 
+  public void clearIngestionKillMessageAndVerify(String clusterName, String versionTopicName) {
+    long startTs = System.currentTimeMillis();
+    ParticipantMessageKey key = new ParticipantMessageKey();
+    key.messageType = ParticipantMessageType.KILL_PUSH_JOB.getValue();
+    key.resourceName = versionTopicName;
+    ParticipantMessageValue value;
+    try {
+      value = RetryUtils.executeWithMaxAttempt(
+          () -> participantStoreClientsManager.getReader(clusterName).get(key).get(),
+          3,
+          Duration.ofSeconds(5),
+          Collections.singletonList(Exception.class));
+      if (value == null) {
+        return;
+      }
+    } catch (Exception e) {
+      LOGGER.error(
+          "Failed to check if kill message exists in participant store for store-version: {} in cluster: {}",
+          versionTopicName,
+          clusterName,
+          e);
+      return;
+    }
+    KillPushJob killPushJobMessage = (KillPushJob) value.messageUnion;
+    LOGGER.info(
+        "Deleting kill message for store-version: {} in cluster: {}. Message age: {}",
+        versionTopicName,
+        clusterName,
+        System.currentTimeMillis() - killPushJobMessage.getTimestamp());
+    deleteParticipantStoreKillMessage(clusterName, versionTopicName);
+
+    // wait for kill message to be removed
```

Comment:
Since `clearIngestionKillMessageAndVerify` will be executed in admin topic (AdminExecutionTask -> admin.addVersionAndStartIngestion). Can we rely on the admin retry and perform an immediate check and fail fast with a `VeniceRetriableException` so while we are waiting for the kill message to be deleted we are not blocking the admin topic from executing other admin messages? This is how we wait for topic creation today.

Reply:
This means we can end up sending multiple clear/delete kill messages unless we do some bookkeeping. Is that right?

Reply:
Right, as we discussed offline I don't think bookkeeping is needed here since this is an edge case and a couple of harmless duplicate delete when handling an edge case is not worth the effort to perform bookkeeping.

Reply:
I've updated PR (we'll spend at most 1min45 secs in clearing KILL message before throwing exception). Could you please take another look? Thanks.